src/dict.c

```diff
@@ -755,8 +758,15 @@ void dictRelease(dict *d)
 {
     /* Someone may be monitoring a dict that started rehashing, before
      * destroying the dict fake completion. */
-    if (dictIsRehashing(d) && d->type->rehashingCompleted)
-        d->type->rehashingCompleted(d);
+    if (dictIsRehashing(d)) {
+        if (d->type->rehashingCompleted)
+            d->type->rehashingCompleted(d);
+        if (d->type->sizeChanged)
+            d->type->sizeChanged(d, -(DICTHT_SIZE(d->ht_size_exp[0]) + DICTHT_SIZE(d->ht_size_exp[1])));
+    } else {
+        if (d->type->sizeChanged)
+            d->type->sizeChanged(d, -DICTHT_SIZE(d->ht_size_exp[0]));
```

Comment:
the mainly reason i added this callback is here, we need to trigger the callback even if the dict is not in the rehashing status.

Reply:
now there is no this issue, right?

Reply:
yes, now we will subtract the current size of buckets before releasing or clearing.